client/src/sw.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017 The Absolute Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface ExtendableEvent extends Event {
```

Comment:
Do we need to have this type(built-in type)? I'm not sure if it is a right way.
(I don't have an deep experience in typescript, I'll also ask it to another people.)

Reply:
There are already built-in types in TypeScript.
https://github.com/Microsoft/TypeScript/blob/master/lib/lib.webworker.d.ts

I think you can just use them :)